girder/api/rest.py

```diff
@@ -914,35 +915,34 @@ def requireParams(self, required, provided):
         :param provided: The list of provided parameters.
         :type provided: dict
         """
-        if isinstance(required, six.string_types):
-            required = (required,)
+        if provided is None and isinstance(required, dict):
+            for name, val in six.viewitems(required):
+                if val is None:
+                    raise RestException('Parameter "%s" is required.' % name)
+        else:
+            if isinstance(required, six.string_types):
+                required = (required,)
 
-        for param in required:
-            if param not in provided:
-                raise RestException("Parameter '%s' is required." % param)
+            for param in required:
+                if provided is None or param not in provided:
+                    raise RestException('Parameter "%s" is required.' % param)
 
-    def boolParam(self, key, params, default=None):
+    @staticmethod
```

Comment:
When reading through the class, why are these two (this and `getPagingParams`) `staticmethod` but no other methods, this smells a bit off.  At least a comment on each of these explaining why would help.

Also, in the `boundHandler` docstring at the bottom of this module, it has the following text which needs to be updated with this `staticmethod` change.

    this to gain access to bound convenience methods like ``self.model``,
        ``self.boolParam``, ``self.requireParams``, etc.